sound/soc/soc-topology.c

```diff
@@ -482,10 +482,10 @@ static void remove_widget(struct snd_soc_component *comp,
 
 			snd_ctl_remove(card, kcontrol);
 
-			kfree(dobj->control.dvalues);
+			kfree(se->dobj.control.dvalues);
 			for (j = 0; j < se->items; j++)
-				kfree(dobj->control.dtexts[j]);
-			kfree(dobj->control.dtexts);
+				kfree(se->dobj.control.dtexts[j]);
+			kfree(se->dobj.control.dtexts);
```

Comment:
help me understand something here: the existing code pattern is repeated a number of times, e.g. it's also used for enums:
kfree(dobj->control.dvalues);
	for (i = 0; i < se->items; i++)
		kfree(dobj->control.dtexts[i]);
	kfree(dobj->control.dtexts);

so either we fix this everywhere or we don't. I feel uncomfortable with a specific change that breaks a mostly regular pattern so far.

Reply:
@plbossart i cant be changed everywhere. In the remove_enum() method, the dbobj is actually the control's dobj. whereas in remove_widget, the inpur argument dobj is the widget's dobj and what we're trying to delete is the control's dobj's texts and values. 
It was a genuine null pointer dereference I saw with the legacy driver that led to me discovering this error while trying to unload the snd_soc_skl module. We dont have enum controls in SOF. So we never encountered this problem before.

Reply:
@ranj063 then you'd really want a comment telling reviewers/users of that code that this is very specific and intentional with explanations

Reply:
@plbossart updated this now. I know the change isnt quite obvious but it is a serious error and hopefully the comment clarifies it.